--- extras/posix/posix.open.m.ORIG	2014-08-05 01:07:42.710423394 +0000
+++ extras/posix/posix.open.m	2014-08-05 01:17:21.687241227 +0000
@@ -71,7 +71,7 @@
     open0(PathName::in, Flags::in, FileDes::out, IO0::di, IO::uo),
     [promise_pure, will_not_call_mercury, thread_safe, tabled_for_io],
  "
-    FileDes = open(PathName, Flags);
+    FileDes = open(PathName, Flags, 0);
     IO = IO0;
 ").
 
