--- units_3_7.ORIG/units.gpr	2009-05-27 13:26:56.000000000 +0000
+++ units_3_7/units.gpr	2017-05-12 21:13:34.523632665 +0000
@@ -3,6 +3,21 @@
 
 Project Units is
 
+   type Development_Type is ("Debug", "Release");
+   Development : Development_Type := external ("Development", "Release");
+   
+   type Legacy_Type is ("Ada95", "Ada2005", "Ada2012");
+   Legacy : Legacy_Type := external ("Legacy", "Ada2005");
+
+   type Library_T is ("static", "static-pic", "relocatable");
+   Library_Type : Library_T := external ("LIBRARY_TYPE", "static");
+   
+   for Library_Name use "@LIB_NAME@";
+   for Library_Version use "lib@LIB_NAME@@SO_EXTENSION@";
+   for Library_Kind use Library_Type;
+   for Object_Dir use "obj/" & Library_Type;
+   for Library_Dir use "lib/" & Library_Type;
+   
    for Source_Dirs use (".");
    for Source_Files use 
        (  "float_measures.ads",
@@ -39,23 +54,60 @@
        );
 
    package Binder is
-      for Default_Switches ("ada") use
-         Strings_Edit.Binder'Default_Switches ("ada");
+      case Development is
+         when "Debug" =>
+            for Default_Switches ("ada") use ("-E");
+         when "Release" =>
+            for Default_Switches ("ada") use ();
+      end case;
    end Binder;
 
    package Builder is
-      for Default_Switches ("ada") use
-         Strings_Edit.Builder'Default_Switches ("ada");
+      case Development is
+         when "Debug" =>
+            for Default_Switches ("ada") use ("-g");
+         when "Release" =>
+            for Default_Switches ("ada") use ("-s");
+      end case;
    end Builder;
 
    package Compiler is
-      for Default_Switches ("ada") use
-         Strings_Edit.Compiler'Default_Switches ("ada");
+      case Development is
+         when "Debug" =>
+            case Legacy is
+               when "Ada2012" =>
+                  for Default_Switches ("ada") use
+                     ("-gnatwcikmoprvz.c.p.t.w.x", "-gnato", "-gnatVa", "-gnatf", "-g", "-fstack-check", "-gnat12");
+               when "Ada2005" =>
+                  for Default_Switches ("ada") use
+                     ("-gnatwcikmoprvZ.c.p.t.w.x", "-gnato", "-gnatVa", "-gnatf", "-g", "-fstack-check", "-gnat05");
+               when "Ada95" =>
+                  for Default_Switches ("ada") use
+                     ("-gnatwcikmoprvz.c.p.t.w.x", "-gnato", "-gnatVa", "-gnatf", "-g", "-fstack-check", "-gnat95");
+            end case;
+         when "Release" =>
+            case Legacy is
+               when "Ada2012" =>
+                  for Default_Switches ("ada") use
+                     ("-gnatwcikmoprvz.c.p.t.w.x", "-gnatp", "-gnatVa", "-gnatf", "-O2", "-gnatn", "-gnat12");
+               when "Ada2005" =>
+                  for Default_Switches ("ada") use
+                     ("-gnatwcikmoprvz.c.p.t.w.x", "-gnatp", "-gnatVa", "-gnatf", "-O2", "-gnatn", "-gnat05");
+               when "Ada95" =>
+                  for Default_Switches ("ada") use
+                     ("-gnatwcikmoprvz.c.p.t.w.x", "-gnatp", "-gnatVa", "-gnatf", "-O2", "-gnatn", "-gnat95");
+            end case;
+      end case;
+      for Default_Switches ("ada") use Compiler'Default_Switches ("ada") & Split (external ("ADAFLAGS", ""), " ");
    end Compiler;
 
    package Linker is
-      for Default_Switches ("ada") use
-         Strings_Edit.Linker'Default_Switches ("ada");
+      case Development is
+         when "Debug" =>
+            for Default_Switches ("ada") use ("-g");
+         when "Release" =>
+            for Default_Switches ("ada") use ("-O2", "-s");
+      end case;
    end Linker;
 
-end Units;
\ No newline at end of file
+end Units;
