diff -urN mercury-srcdist-13.05.2.orig/extras/posix/Mmakefile mercury-srcdist-13.05.2/extras/posix/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/posix/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/posix/Mmakefile	2014-02-18 11:21:22.000000000 +0100
@@ -34,7 +34,7 @@
 
 .PHONY: install
 install: libposix.install
-	cp $(ADDITIONAL_HDRS) $(INSTALL_PREFIX)/lib/mercury/inc
+	cp $(ADDITIONAL_HDRS) $(INSTALL_INC_DIR)
 
 .PHONY: clean
 clean:
