diff -urN mercury-srcdist-13.05.2.orig/extras/solver_types/library/any_map.m mercury-srcdist-13.05.2/extras/solver_types/library/any_map.m
--- mercury-srcdist-13.05.2.orig/extras/solver_types/library/any_map.m	2013-04-09 11:22:17.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/solver_types/library/any_map.m	2014-02-18 11:21:53.000000000 +0100
@@ -529,8 +529,6 @@
 :- import_module pair.
 :- import_module require.
 :- import_module string.
-:- import_module svmap.
-:- import_module svset.
 
 :- type any_map(K, V)   ==  any_tree234(K, V).
 
