diff -urN mercury-srcdist-13.05.2.orig/extras/base64/Mmakefile mercury-srcdist-13.05.2/extras/base64/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/base64/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/base64/Mmakefile	2014-02-18 11:20:11.000000000 +0100
@@ -23,7 +23,7 @@
 
 .PHONY: install
 install:
-	$(MAKE) install INSTALL_PREFIX=$(INSTALL_PREFIX)/extras
+	$(MAKE) install
 
 .PHONY: realclean
 realclean:
diff -urN mercury-srcdist-13.05.2.orig/extras/fixed/Mmakefile mercury-srcdist-13.05.2/extras/fixed/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/fixed/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/fixed/Mmakefile	2014-02-18 11:20:11.000000000 +0100
@@ -13,7 +13,7 @@
 
 .PHONY: build
 build:
-	$(MAKE) build INSTALL_PREFIX=$(INSTALL_PREFIX)/extras
+	$(MAKE) build
 
 # The depend target does nothing, mmc --make will handle that for us.
 #
@@ -22,7 +22,7 @@
 
 .PHONY: install
 install:
-	$(MAKE) install INSTALL_PREFIX=$(INSTALL_PREFIX)/extras
+	$(MAKE) install
 
 .PHONY: realclean
 realclean:
diff -urN mercury-srcdist-13.05.2.orig/extras/references/Mmakefile mercury-srcdist-13.05.2/extras/references/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/references/Mmakefile	2013-04-09 11:22:17.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/references/Mmakefile	2014-02-18 11:20:11.000000000 +0100
@@ -13,7 +13,7 @@
 
 .PHONY: build
 build:
-	$(MAKE) build INSTALL_PREFIX=$(INSTALL_PREFIX)/extras
+	$(MAKE) build
 
 # The depend target does nothing, mmc --make will handle that for us.
 #
@@ -22,7 +22,7 @@
 
 .PHONY: install
 install:
-	$(MAKE) install INSTALL_PREFIX=$(INSTALL_PREFIX)/extras
+	$(MAKE) install
 
 .PHONY: realclean
 realclean:
