diff -urN mercury-srcdist-13.05.2.orig/extras/cgi/Mmakefile mercury-srcdist-13.05.2/extras/cgi/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/cgi/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/cgi/Mmakefile	2014-02-18 11:20:46.000000000 +0100
@@ -12,6 +12,9 @@
 
 MAIN_TARGET = $(LIBRARY)
 
+LIBGRADES := $(filter-out java,$(LIBGRADES))
+LIBGRADES := $(filter-out erlang,$(LIBGRADES))
+
 MCFLAGS-mercury_www = --no-warn-nothing-exported
 
 depend: mercury_www.depend
diff -urN mercury-srcdist-13.05.2.orig/extras/complex_numbers/Mmakefile mercury-srcdist-13.05.2/extras/complex_numbers/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/complex_numbers/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/complex_numbers/Mmakefile	2014-02-18 11:20:46.000000000 +0100
@@ -10,6 +10,9 @@
 
 MAIN_TARGET = all
 
+LIBGRADES := $(filter-out java,$(LIBGRADES))
+LIBGRADES := $(filter-out erlang,$(LIBGRADES))
+
 all:	libcomplex_numbers
 
 depend:	complex_numbers.depend
diff -urN mercury-srcdist-13.05.2.orig/extras/curs/Mmakefile mercury-srcdist-13.05.2/extras/curs/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/curs/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/curs/Mmakefile	2014-02-18 11:20:46.000000000 +0100
@@ -26,7 +26,8 @@
 # Omit this line if you want to install the default grades.
 # Edit this line if you want to install with different grades.
 #
-#LIBGRADES = asm_fast.gc hlc.gc
+LIBGRADES := $(filter-out java,$(LIBGRADES))
+LIBGRADES := $(filter-out erlang,$(LIBGRADES))
 
 # The following standard libraries must be linked against in any
 # application using the curs library (i.e. MLLIBS must include
diff -urN mercury-srcdist-13.05.2.orig/extras/curses/Mmakefile mercury-srcdist-13.05.2/extras/curses/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/curses/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/curses/Mmakefile	2014-02-18 11:20:46.000000000 +0100
@@ -13,6 +13,9 @@
 # this line to include the appropriate curses library for your OS. 
 MLLIBS = -lncurses
 
+LIBGRADES := $(filter-out java,$(LIBGRADES))
+LIBGRADES := $(filter-out erlang,$(LIBGRADES))
+
 -include ../Mmake.params
 
 default_target:	libmcurses
diff -urN mercury-srcdist-13.05.2.orig/extras/dynamic_linking/Mmakefile mercury-srcdist-13.05.2/extras/dynamic_linking/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/dynamic_linking/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/dynamic_linking/Mmakefile	2014-02-18 11:20:46.000000000 +0100
@@ -17,6 +17,9 @@
 CFLAGS = -g
 MLFLAGS += --no-strip
 
+LIBGRADES := $(filter-out java,$(LIBGRADES))
+LIBGRADES := $(filter-out erlang,$(LIBGRADES))
+
 -include ../Mmake.params
 
 main_target: libdl
diff -urN mercury-srcdist-13.05.2.orig/extras/graphics/mercury_glut/Mmakefile mercury-srcdist-13.05.2/extras/graphics/mercury_glut/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/graphics/mercury_glut/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/graphics/mercury_glut/Mmakefile	2014-02-18 11:20:46.000000000 +0100
@@ -10,6 +10,9 @@
 
 MAIN_TARGET = libmercury_glut
 
+LIBGRADES := $(filter-out java,$(LIBGRADES))
+LIBGRADES := $(filter-out erlang,$(LIBGRADES))
+
 # The following libraries are for X on Linux (Debian) using Mesa.
 	
 # Libaries required by X windows.
diff -urN mercury-srcdist-13.05.2.orig/extras/graphics/mercury_opengl/Mmakefile mercury-srcdist-13.05.2/extras/graphics/mercury_opengl/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/graphics/mercury_opengl/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/graphics/mercury_opengl/Mmakefile	2014-02-18 11:20:46.000000000 +0100
@@ -15,6 +15,9 @@
 
 MAIN_TARGET = libmercury_opengl
 
+LIBGRADES := $(filter-out java,$(LIBGRADES))
+LIBGRADES := $(filter-out erlang,$(LIBGRADES))
+
 # Specify what libraries we need to link against for OpenGL on this system.
 # (The following works on Debian with mesa as the OpenGL implementation).
 GL_LIBS = -lGL -lGLU -L/usr/X11R6/lib -lX11 -lXt -lICE -lXext -lSM
diff -urN mercury-srcdist-13.05.2.orig/extras/graphics/mercury_tcltk/Mmakefile mercury-srcdist-13.05.2/extras/graphics/mercury_tcltk/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/graphics/mercury_tcltk/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/graphics/mercury_tcltk/Mmakefile	2014-02-18 11:20:46.000000000 +0100
@@ -21,6 +21,9 @@
 
 MAIN_TARGET = $(LIBRARY)
 
+LIBGRADES := $(filter-out java,$(LIBGRADES))
+LIBGRADES := $(filter-out erlang,$(LIBGRADES))
+
 depend: mercury_tcltk.depend
 
 install: libmercury_tcltk.install
diff -urN mercury-srcdist-13.05.2.orig/extras/lex/Mmakefile mercury-srcdist-13.05.2/extras/lex/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/lex/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/lex/Mmakefile	2014-02-18 11:20:46.000000000 +0100
@@ -4,6 +4,9 @@
 # Public License - see the file COPYING.LIB in the Mercury distribution.
 #-----------------------------------------------------------------------------#
 
+LIBGRADES := $(filter-out java,$(LIBGRADES))
+LIBGRADES := $(filter-out erlang,$(LIBGRADES))
+
 # This is file exists so that do an `mmake' from the top-level of the extras
 # distribution works.  The actual work of building this library is done
 # using the Makefile and mmc --make.  This file just fowards all the work
diff -urN mercury-srcdist-13.05.2.orig/extras/moose/Mmakefile mercury-srcdist-13.05.2/extras/moose/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/moose/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/moose/Mmakefile	2014-02-18 11:20:46.000000000 +0100
@@ -6,6 +6,8 @@
 
 # GRADE = asm_fast.gc.debug
 # GRADE = asm_fast.gc.prof
+LIBGRADES := $(filter-out java,$(LIBGRADES))
+LIBGRADES := $(filter-out erlang,$(LIBGRADES))
 
 # MCFLAGS = -O6
 # MCFLAGS = --intermodule-optimization -O6
diff -urN mercury-srcdist-13.05.2.orig/extras/odbc/Mmakefile mercury-srcdist-13.05.2/extras/odbc/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/odbc/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/odbc/Mmakefile	2014-02-18 11:20:46.000000000 +0100
@@ -8,6 +8,8 @@
 
 #-----------------------------------------------------------------------------#
 # Configuration
+LIBGRADES := $(filter-out java,$(LIBGRADES))
+LIBGRADES := $(filter-out erlang,$(LIBGRADES))
 
 # The driver manager.
 # Legal values for MODBC_DRIVER are MODBC_IODBC, MODBC_UNIX, and MODBC_MS.
diff -urN mercury-srcdist-13.05.2.orig/extras/posix/Mmakefile mercury-srcdist-13.05.2/extras/posix/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/posix/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/posix/Mmakefile	2014-02-18 11:20:46.000000000 +0100
@@ -8,6 +8,9 @@
 
 -include ../Mmake.params
 
+LIBGRADES := $(filter-out java,$(LIBGRADES))
+LIBGRADES := $(filter-out erlang,$(LIBGRADES))
+
 # The following is needed (on some systems) to enable declarations of
 # Posix functions in standard C header files.
 MGNUCFLAGS = --no-ansi
diff -urN mercury-srcdist-13.05.2.orig/extras/solver_types/library/Mmakefile mercury-srcdist-13.05.2/extras/solver_types/library/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/solver_types/library/Mmakefile	2013-04-09 11:22:17.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/solver_types/library/Mmakefile	2014-02-18 11:20:46.000000000 +0100
@@ -16,6 +16,9 @@
 
 install: libany.install
 
+LIBGRADES := $(filter-out java,$(LIBGRADES))
+LIBGRADES := $(filter-out erlang,$(LIBGRADES))
+
 MCFLAGS-any= --no-warn-nothing-exported --no-warn-interface-imports
 
 .PHONY: check
diff -urN mercury-srcdist-13.05.2.orig/extras/xml/Mmakefile mercury-srcdist-13.05.2/extras/xml/Mmakefile
--- mercury-srcdist-13.05.2.orig/extras/xml/Mmakefile	2013-04-09 11:22:17.000000000 +0200
+++ mercury-srcdist-13.05.2/extras/xml/Mmakefile	2014-02-18 11:20:46.000000000 +0100
@@ -8,6 +8,9 @@
 
 INSTALL_PREFIX := $(INSTALL_PREFIX)/extras
 
+LIBGRADES := $(filter-out java,$(LIBGRADES))
+LIBGRADES := $(filter-out erlang,$(LIBGRADES))
+
 .PHONY: default_target
 default_target: tryit libxml
 
