--- gtkada-2016_p20170429220214/Makefile.in.ORIG	2017-04-29 23:52:03.792119739 +0000
+++ gtkada-2016_p20170429220214/Makefile.in	2017-04-29 23:52:19.544526899 +0000
@@ -118,9 +118,9 @@
 install: install-clean
 	${GPRINSTALL_FULL} -XLIBRARY_TYPE=$(LIBRARY_TYPE_FOR_TOOLS) \
 		--mode=usage -Psrc/tools/tools.gpr
-	cd testgtk; ${GPRINSTALL_FULL} -XLIBRARY_TYPE=$(LIBRARY_TYPE_FOR_TOOLS) \
-		--mode=usage --exec-subdir=${DESTDIR}${exampledir} \
-		-Ptestgtk.gpr -aP ../src
+#	cd testgtk; ${GPRINSTALL_FULL} -XLIBRARY_TYPE=$(LIBRARY_TYPE_FOR_TOOLS) \
+#		--mode=usage --exec-subdir=${DESTDIR}${exampledir} \
+#		-Ptestgtk.gpr -aP ../src
 
 	@echo '-----------------------------------------------------------------------'
 	@echo '--  GtkAda has now been installed.                                   --'
