--- dbus.ORIG/dbus.scm	2021-01-12 15:39:55.814042219 +0000
+++ dbus/dbus.scm	2021-01-12 15:49:14.806151186 +0000
@@ -100,7 +100,7 @@
 ;; have a use for the "variant" concept; but since dbus has a variant type,
 ;; we need a way of representing one when preparing a message for marshalling.
 (define-record-type variant
-  (make-variant #!optional data)
+  (make-variant data)
   variant?
   (data variant-data))
 
