--- distro-info-0.10/Makefile~	2012-05-28 22:41:33.000000000 +0000
+++ distro-info-0.10/Makefile	2013-07-16 21:39:48.803953479 +0000
@@ -29,8 +29,8 @@
 	install -m 644 $(wildcard doc/*.1) $(DESTDIR)$(PREFIX)/share/man/man1
 	install -d $(DESTDIR)$(PREFIX)/share/perl5/Debian
 	install -m 644 $(wildcard perl/Debian/*.pm) $(DESTDIR)$(PREFIX)/share/perl5/Debian
-	cd python && python setup.py install --root="$(DESTDIR)" --no-compile --install-layout=deb
-	cd python && python3 setup.py install --root="$(DESTDIR)" --no-compile --install-layout=deb
+	cd python && python setup.py install --root="$(DESTDIR)" --no-compile #--install-layout=deb
+	cd python && python3 setup.py install --root="$(DESTDIR)" --no-compile #--install-layout=deb
 
 test: test-commandline test-perl test-python
 
