--- ssax-sxml.ORIG/libs/gambit/myenv.sch	2020-05-17 15:06:24.399146149 +0000
+++ ssax-sxml/libs/gambit/myenv.sch	2020-05-17 15:07:11.270711146 +0000
@@ -28,16 +28,16 @@
 ; correct, but more portable (and less efficient)
 
 ; Mutable increment
-(define-macro (++! x) `(set! ,x (##fixnum.+ 1 ,x)))
+(define-macro (++! x) `(set! ,x (fx+ 1 ,x)))
 
 ; Read-only increment
-(define-macro (++ x) `(##fixnum.+ 1 ,x))
+(define-macro (++ x) `(fx+ 1 ,x))
 
 ; Mutable decrement
-(define-macro (--! x) `(set! ,x (##fixnum.- ,x 1)))
+(define-macro (--! x) `(set! ,x (fx- ,x 1)))
 
 ; Read-only decrement
-(define-macro (-- x) `(##fixnum.- ,x 1))
+(define-macro (-- x) `(fx- ,x 1))
 
 ;==============================================================================
 ; Some useful control operators
@@ -200,5 +200,5 @@
 ; DL: taken from previous versions of "myenv.scm"
 
 ; DL: aliases for ++ and --
-(define-macro (inc x) `(##fixnum.+ 1 ,x))
-(define-macro (dec x) `(##fixnum.- ,x 1))
+(define-macro (inc x) `(fx+ 1 ,x))
+(define-macro (dec x) `(fx- ,x 1))
