--- bc-20160321.ORIG/./bc.gpr	2017-05-04 16:15:15.672625194 +0000
+++ bc-20160321/./bc.gpr	2017-05-04 19:29:30.647768717 +0000
@@ -12,12 +12,12 @@
 --  write to the Free Software Foundation, 59 Temple Place - Suite
 --  330, Boston, MA 02111-1307, USA.
 
-project BC is
+library project BC is
 
    type Language is ("gnat95", "gnat05", "gnat12");
    Chosen_Language : Language := external ("LANGUAGE", "gnat95");
 
-   type Library_T is ("static", "relocatable");
+   type Library_T is ("static", "static-pic", "relocatable");
    Library_Type : Library_T := external ("LIBRARY_TYPE", "static");
 
    Sources :=
@@ -429,37 +429,45 @@
 
    for Library_Name use "bc";
    for Library_Kind use Library_Type;
-   for Library_Dir use "./lib-" & Library_Type;
-   case Library_Type is
-      when "relocatable" =>
-         for Library_Src_Dir use "./include";
-         for Library_Interface use Source_Units;
-      when "static" =>
-         --  Neither gnatmake nor gprbuild handles this case
-         --  properly. Rely on the relocatable build to copy the
-         --  required interface units to ./include.
-         null;
-   end case;
-   for Library_Options use External_As_List ("LDFLAGS", " ");
+   
+--   for Library_Dir use "./lib-" & Library_Type;
+--   case Library_Type is
+--      when "relocatable" =>
+--         for Library_Src_Dir use "./include";
+--         for Library_Interface use Source_Units;
+--      when "static" =>
+--         --  Neither gnatmake nor gprbuild handles this case
+--         --  properly. Rely on the relocatable build to copy the
+--         --  required interface units to ./include.
+--         null;
+--   end case;
+   
+--   for Library_Options use External_As_List ("LDFLAGS", " ");
 
    for Languages use ("ada");
 
    for Source_Dirs use ("src");
    for Source_Files use Sources;
 
-   for Object_Dir use ".build-" & Library_Type;
-
+   for Object_Dir use "obj/" & Library_Type;
+   for Library_Dir use "lib/" & Library_Type;
+   
+   Ada_Flags := Split (external ("ADAFLAGS", ""), " ");
+   
    package Compiler is
-      for Default_Switches ("Ada") use
-        (
+     for Default_Switches ("Ada") use
+       (
          "-" & Chosen_Language,
-         "-g",       --  for debug symbols
-         "-O2",      --  standard optimisation
+--         "-g",       --  for debug symbols
+         "-O2"       --  standard optimisation
+       )
+       & Ada_Flags &
+       (
          "-gnatqQ",  --  to keep ALI files even if there are compilation errors
          "-gnatafo", --  assertions, full reports, integer overflow
          "-gnaty",   --  standard style checks
          "-gnatwaL"  --  all standard warnings except for elaboration
-        );
+       );
    end Compiler;
 
 end BC;
--- bc-20160321.ORIG/./Makefile.in	2017-05-04 16:15:15.678624974 +0000
+++ bc-20160321/./Makefile.in	2017-05-04 19:44:16.586371380 +0000
@@ -15,28 +15,41 @@
 # Template for the top-level (GNU) Makefile for Booch Components. Used
 # for installation and distribution construction.
 
+.NOTPARALLEL:
+
 prefix = @PREFIX@
+LIBDIR = @LIBDIR@
+DESTDIR =
+
 GPRBUILD ?= gprbuild
+GPRINSTALL ?= gprinstall
+
+PROCESSORS = @PROCESSORS@
+
+BC_BUILDER = $(GPRBUILD) -j$(PROCESSORS) -v -R -p -Pbc 
+BC_INSTALLER = $(GPRINSTALL) -v -p -f --build-var=LIBRARY_TYPE \
+	-Pbc --install-name=bc \
+	--prefix="$(DESTDIR)$(prefix)" --link-lib-subdir="$(LIBDIR)"
 
 # 'make' to make the BC libraries for use with bc.gpr.
 # 'make install' to install the BC libraries with your GNAT.
 # 'make dist' to make the distribution.
 
 libs:
-	$(GPRBUILD) -p -Pbc -XLIBRARY_TYPE=static
-	$(GPRBUILD) -p -Pbc -XLIBRARY_TYPE=relocatable
+	$(BC_BUILDER) -XLIBRARY_TYPE=static
+	$(BC_BUILDER) -XLIBRARY_TYPE=static-pic
+	$(BC_BUILDER) -XLIBRARY_TYPE=relocatable
 
 install:
-	-mkdir -p $(prefix)/lib/gnat/
-	cp bc.gpr-for-installation $(prefix)/lib/gnat/bc.gpr
-	-mkdir -p $(prefix)/include/bc
-	cd include; tar -c -f- *.ad[bs] | tar -x -f- -C $(prefix)/include/bc/
-	-mkdir -p $(prefix)/lib/bc/lib-static
-	tar -c -f- lib-static | tar -x -f- -C $(prefix)/lib/bc/
-	chmod -w $(prefix)/lib/bc/lib-static/*.ali
-	-mkdir -p $(prefix)/lib/bc/lib-relocatable
-	tar -c -f- lib-relocatable | tar -x -f- -C $(prefix)/lib/bc/
-	chmod -w $(prefix)/lib/bc/lib-relocatable/*.ali
+	$(BC_INSTALLER) -XLIBRARY_TYPE=static \
+		--lib-subdir=$(LIBDIR)/bc/static \
+		--build-name=static
+	$(BC_INSTALLER) -XLIBRARY_TYPE=static-pic \
+		--lib-subdir=$(LIBDIR)/bc/static-pic \
+		--build-name=static-pic
+	$(BC_INSTALLER) -XLIBRARY_TYPE=relocatable \
+		--lib-subdir=$(LIBDIR)/bc/relocatable \
+		--build-name=relocatable
 
 # Distribution construction
 
