--- xmlada.ORIG/Makefile.in	2017-05-02 21:00:26.167761860 +0000
+++ xmlada/Makefile.in	2017-05-02 22:54:16.743738140 +0000
@@ -1,3 +1,5 @@
+.NOTPARALLEL:
+
 ## The type of library we want to build. Possible values:
 ##   relocatable
 ##   static
@@ -14,10 +16,12 @@
 
 # Set a default value (ignored if set in the environment)
 PROCESSORS?=0
-GPRBUILD=@gprbuild@ -j${PROCESSORS} -m ${GPRBUILD_OPTIONS}
+GPRBUILD=@gprbuild@
+GPRBUILD2=${GPRBUILD} -j${PROCESSORS} -m ${GPRBUILD_OPTIONS}
 GPRINSTALL=@gprinstall@
 TARGET=@target@
 HOST=@host@
+LIBDIR=lib
 
 # Compiler mode: one of "distrib", "Debug", "Production", "profile", "coverage"
 MODE=@BUILD_TYPE@
@@ -34,7 +38,7 @@
 MODULE_INSTALL=${MODULES:%=%_inst}
 
 GPROPTS=-XBUILD=${MODE} -XPROCESSORS=${PROCESSORS}
-PSUB=lib/gnat
+PSUB=share/gpr
 
 ifeq (${HOST},${TARGET})
 IPREFIX=${prefix}
@@ -52,28 +56,28 @@
 endif
 
 static:
-	${GPRBUILD} -p -XLIBRARY_TYPE=static ${GPROPTS} xmlada.gpr
+	${GPRBUILD2} -p -XLIBRARY_TYPE=static ${GPROPTS} xmlada.gpr
 
 shared relocatable:
-	${GPRBUILD} -p -XLIBRARY_TYPE=relocatable ${GPROPTS} xmlada.gpr
+	${GPRBUILD2} -p -XLIBRARY_TYPE=relocatable ${GPROPTS} xmlada.gpr
 
 static-pic:
-	${GPRBUILD} -p -XLIBRARY_TYPE=static-pic ${GPROPTS} xmlada.gpr
+	${GPRBUILD2} -p -XLIBRARY_TYPE=static-pic ${GPROPTS} xmlada.gpr
 
 # Rebuild the tests (and not the XML/Ada code itself)
 test: force
-	${GPRBUILD} -p -XLIBRARY_TYPE=static ${GPROPTS} \
+	${GPRBUILD2} -p -XLIBRARY_TYPE=static ${GPROPTS} \
 		-XTESTS_ACTIVATED=Only xmlada.gpr
 
 # Run the tests. This requires an install of XML/Ada first
 run_test: force
-	@make --no-print-directory -C tests tests
+	+@make --no-print-directory -C tests tests
 
 run_test_valgrind: test
 	cd tests/domt; valgrind --tool=memcheck --leak-check=full ./obj/testxml -auto
 
 docs: force
-	-${MAKE} -e -C docs html latexpdf
+	+$(MAKE) -e -C docs html latexpdf
 
 install-clean-obsolete:
 ifneq (,$(wildcard $(IPREFIX)/$(PSUB)/manifests/xmlada_schema))
@@ -100,18 +104,22 @@
 	${GPRINSTALL} -f -p -XLIBRARY_TYPE=static ${GPROPTS} \
 		--prefix=${IPREFIX} --project-subdir=$(PSUB) \
 		--build-var=XMLADA_BUILD --build-name=static \
+		--lib-subdir=$(LIBDIR)/xmlada/static \
 		--install-name=xmlada xmlada.gpr
 
 install-shared install-relocatable:
 	${GPRINSTALL} -f -p -XLIBRARY_TYPE=relocatable ${GPROPTS} \
 		--prefix=${IPREFIX} --project-subdir=$(PSUB) \
 		--build-var=XMLADA_BUILD --build-name=relocatable \
+		--lib-subdir=$(LIBDIR)/xmlada/relocatable \
+		--link-lib-subdir=$(LIBDIR) \
 		--install-name=xmlada xmlada.gpr
 
 install-static-pic:
 	${GPRINSTALL} -f -p -XLIBRARY_TYPE=static-pic ${GPROPTS} \
 		--prefix=${IPREFIX} --project-subdir=$(PSUB) \
 		--build-var=XMLADA_BUILD --build-name=static-pic \
+		--lib-subdir=$(LIBDIR)/xmlada/static-pic \
 		--install-name=xmlada xmlada.gpr
 
 
@@ -120,7 +128,7 @@
 	-gprclean -XLIBRARY_TYPE=$* -q -XTESTS_ACTIVATED=Yes xmlada.gpr
 
 clean: clean_relocatable clean_static clean_static-pic
-	cd docs; ${MAKE} -e clean
+	+$(MAKE) -C docs -e clean
 
 distclean: clean
 	${RM} tests/dom/default.gpr
--- xmlada.ORIG/input_sources/xmlada_input.gpr	2017-05-02 21:00:26.261758322 +0000
+++ xmlada/input_sources/xmlada_input.gpr	2017-05-02 21:01:10.138106824 +0000
@@ -41,7 +41,7 @@
    package Install is
       for Artifacts ("share/doc/xmlada")
         use ("../docs/_build/html/*", "../docs/_build/latex/XMLAda.pdf");
-      for Artifacts ("lib/gnat") use ("../distrib/xmlada.gpr");
+      for Artifacts ("share/gpr") use ("../distrib/xmlada.gpr");
    end Install;
 
 end XmlAda_Input;
--- xmlada.ORIG/docs/Makefile	2017-05-02 21:00:26.145762687 +0000
+++ xmlada/docs/Makefile	2017-05-02 22:55:21.259309785 +0000
@@ -100,7 +100,7 @@
 latexpdf:
 	$(SPHINXBUILD) -b latex $(ALLSPHINXOPTS) $(BUILDDIR)/latex
 	@echo "Running LaTeX files through pdflatex..."
-	make -C $(BUILDDIR)/latex all-pdf
+	+$(MAKE) -C $(BUILDDIR)/latex all-pdf
 	@echo "pdflatex finished; the PDF files are in $(BUILDDIR)/latex."
 
 text:
