--- strings.ORIG/./strings_edit.gpr	2017-05-07 23:55:37.103742532 +0000
+++ strings/./strings_edit.gpr	2017-05-08 00:00:26.860161307 +0000
@@ -1,14 +1,21 @@
 project Strings_Edit is
 
    type Development_Type is ("Debug", "Release");
-   Development : Development_Type := external ("Development", "Debug");
+   Development : Development_Type := external ("Development", "Release");
    
    type Legacy_Type is ("Ada95", "Ada2005", "Ada2012");
    Legacy : Legacy_Type := external ("Legacy", "Ada2005");
 
+   type Library_T is ("static", "static-pic", "relocatable");
+   Library_Type : Library_T := external ("LIBRARY_TYPE", "static");
+   
+   for Library_Name use "@LIB_NAME@";
+   for Library_Version use "@SO_NAME@";
+   for Library_Kind use Library_Type;
+   for Object_Dir use "obj/" & Library_Type;
+   for Library_Dir use "lib/" & Library_Type;
+      
    for Source_Dirs use (".");
-   for Object_Dir use ".";
-
    for Source_Files use
        (  "strings_edit-base64.adb",
 	      "strings_edit-base64.ads",
@@ -82,15 +89,6 @@
       end case;
    end Binder;
 
-   package Builder is
-      case Development is
-         when "Debug" =>
-            for Default_Switches ("ada") use ("-g");
-         when "Release" =>
-            for Default_Switches ("ada") use ("-s");
-      end case;
-   end Builder;
-
    package Compiler is
       case Development is
          when "Debug" =>
@@ -118,16 +116,8 @@
                      ("-gnatwcikmoprvz.c.p.t.w.x", "-gnatp", "-gnatVa", "-gnatf", "-O2", "-gnatn", "-gnat95");
             end case;
       end case;
+      for Switches ("ada") use Compiler'Switches ("ada") & Split (external ("ADAFLAGS", ""), " ");
    end Compiler;
 
-   package Linker is
-      case Development is
-         when "Debug" =>
-            for Default_Switches ("ada") use ("-g");
-         when "Release" =>
-            for Default_Switches ("ada") use ("-O2", "-s");
-      end case;
-   end Linker;
-
 end Strings_Edit;
 
