--- devscripts-2.12.6.ORIG/Makefile	2012-06-17 17:37:11.000000000 +0000
+++ devscripts-2.12.6/Makefile	2013-07-16 03:45:25.875599333 +0000
@@ -11,7 +11,7 @@
 DOCDIR ?= $(PREFIX)/share/doc/devscripts
 MAN1DIR ?= $(PREFIX)/share/man/man1
 
-all: version make_scripts $(EXAMPLES) translated_manpages
+all: version make_scripts $(EXAMPLES) #translated_manpages
 
 version:
 	rm -f version
--- devscripts-2.12.6.ORIG/scripts/Makefile	2012-06-21 22:39:10.000000000 +0000
+++ devscripts-2.12.6/scripts/Makefile	2013-07-16 06:49:54.005962335 +0000
@@ -61,7 +61,8 @@
 	pod2man --utf8 --center=" " --release="Debian Utilities" $< > $@
 %.1: %.dbk
 	xsltproc --nonet -o $@ \
-	  /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl $<
+		/usr/share/sgml/docbook/xsl-stylesheets/manpages/docbook.xsl $<
+#	  /usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl $<
 
 # There is a slight chance this gets called twice, once here from here and once
 # from ../po4a/Makefile. Treat files with care.
@@ -91,7 +92,7 @@
 	$(foreach python,$(shell pyversions -r ../debian/control),$(python) setup.py test$(\n))
 
 install: all
-	python setup.py install --root="$(DESTDIR)" --no-compile --install-layout=deb
+	python setup.py install --root="$(DESTDIR)" --no-compile #--install-layout=deb
 	install -dD $(DESTDIR)$(BINDIR) $(DESTDIR)$(LIBDIR)
 	cp $(SCRIPTS) $(DESTDIR)$(BINDIR)
 	ln -sf edit-patch $(DESTDIR)$(BINDIR)/add-patch
