diff -Nuar a/makefile_original_install b/makefile_original_install
--- a/makefile_original_install	2013-10-01 17:52:03.000000000 +0200
+++ b/makefile_original_install	2014-01-20 13:04:41.569999708 +0100
@@ -31,7 +31,10 @@
 	-install -d $(datadir)/newlisp/modules
 	-install -d $(datadir)/newlisp/util
 	-install -d $(datadir)/doc/newlisp
-	-rm $(bindir)/newlisp
+		-install -d $(datadir)/man
+	-install -d $(datadir)/man/man1
+	-install -d $(bindir)
+
 	-install -m 755  newlisp $(bindir)/newlisp-$(VERSION)
 	-ln -s $(bindir)/newlisp-$(VERSION) $(bindir)/newlisp
 	-install -m 755 util/newlispdoc $(bindir)/newlispdoc
