diff -ur mercury-srcdist-13.05.2.orig/library/Mmakefile mercury-srcdist-13.05.2/library/Mmakefile
--- mercury-srcdist-13.05.2.orig/library/Mmakefile	2013-04-09 11:22:17.000000000 +0200
+++ mercury-srcdist-13.05.2/library/Mmakefile	2014-02-18 10:19:48.000000000 +0100
@@ -258,8 +258,8 @@
 	[ -d jmercury/runtime ] || cp -r ../java/runtime jmercury
 
 install_mer_rt: $(RT_LIB_NAME).jar
-	mkdir -p $(INSTALL_PREFIX)/lib/mercury/lib/$(GRADE)
-	cp $(RT_LIB_NAME).jar $(INSTALL_PREFIX)/lib/mercury/lib/$(GRADE)
+	mkdir -p $(INSTALL_LIBDIR)/lib/$(GRADE)
+	cp $(RT_LIB_NAME).jar $(INSTALL_LIBDIR)/lib/$(GRADE)
 
 else
 
diff -ur mercury-srcdist-13.05.2.orig/scripts/Mmake.vars.in mercury-srcdist-13.05.2/scripts/Mmake.vars.in
--- mercury-srcdist-13.05.2.orig/scripts/Mmake.vars.in	2013-04-09 11:22:17.000000000 +0200
+++ mercury-srcdist-13.05.2/scripts/Mmake.vars.in	2014-02-18 10:19:48.000000000 +0100
@@ -684,24 +684,24 @@
 
 INSTALL_PREFIX		= $(DESTDIR_AND_SLASH)@prefix@
 INSTALL_BINDIR		= $(INSTALL_PREFIX)/bin
-INSTALL_LIBDIR		= $(INSTALL_PREFIX)/lib/mercury
+INSTALL_LIBDIR		= $(INSTALL_PREFIX)/@libdir@/mercury
 INSTALL_INFO_DIR	= $(INSTALL_PREFIX)/info
-INSTALL_DVI_DIR		= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_TEXT_DIR	= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_PS_DIR		= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_PDF_DIR		= $(INSTALL_PREFIX)/lib/mercury/doc
+INSTALL_DVI_DIR		= $(INSTALL_PREFIX)/@libdir@/mercury/doc
+INSTALL_TEXT_DIR	= $(INSTALL_PREFIX)/@libdir@/mercury/doc
+INSTALL_PS_DIR		= $(INSTALL_PREFIX)/@libdir@/mercury/doc
+INSTALL_PDF_DIR		= $(INSTALL_PREFIX)/@libdir@/mercury/doc
 INSTALL_MAN_DIR		= $(INSTALL_PREFIX)/man
-INSTALL_HTML_DIR	= $(INSTALL_PREFIX)/lib/mercury/html
-INSTALL_MDB_DOC_DIR	= $(INSTALL_PREFIX)/lib/mercury/mdb
-INSTALL_ELISP_DIR	= $(INSTALL_PREFIX)/lib/mercury/elisp
+INSTALL_HTML_DIR	= $(INSTALL_PREFIX)/@libdir@/mercury/html
+INSTALL_MDB_DOC_DIR	= $(INSTALL_PREFIX)/@libdir@/mercury/mdb
+INSTALL_ELISP_DIR	= $(INSTALL_PREFIX)/@libdir@/mercury/elisp
 INSTALL_CGI_DIR		= $(DESTDIR_AND_SLASH)@CGIDIR@
 
 FINAL_INSTALL_PREFIX	= @prefix@
 FINAL_INSTALL_BINDIR 	= $(FINAL_INSTALL_PREFIX)/bin
-FINAL_INSTALL_LIBDIR	= $(FINAL_INSTALL_PREFIX)/lib/mercury
+FINAL_INSTALL_LIBDIR	= $(FINAL_INSTALL_PREFIX)/@libdir@/mercury
 FINAL_INSTALL_INFO_DIR	= $(FINAL_INSTALL_PREFIX)/info
 FINAL_INSTALL_MAN_DIR	= $(FINAL_INSTALL_PREFIX)/man
-FINAL_INSTALL_ELISP_DIR	= $(FINAL_INSTALL_PREFIX)/lib/mercury/elisp
+FINAL_INSTALL_ELISP_DIR	= $(FINAL_INSTALL_PREFIX)/@libdir@/mercury/elisp
 FINAL_INSTALL_CGI_DIR	= @CGIDIR@
 
 # You should not need to override anything below here
