diff -ur mercury-srcdist-13.05.2.orig/boehm_gc/Mmakefile mercury-srcdist-13.05.2/boehm_gc/Mmakefile
--- mercury-srcdist-13.05.2.orig/boehm_gc/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/boehm_gc/Mmakefile	2014-02-18 10:11:16.000000000 +0100
@@ -190,6 +190,9 @@
 
 # No need to install the Boehm collector for grades which don't use it.
 
+.PHONY: libgrade
+libgrade:
+
 .PHONY: install_headers
 install_headers:
 
@@ -201,6 +204,9 @@
 
 else
 
+.PHONY: libgrade
+libgrade: lib$(GC_GRADE).$A lib$(GC_GRADE).$(EXT_FOR_SHARED_LIB)
+
 .PHONY: install_headers
 install_headers: install_dirs
 	cp $(HEADERS) $(INSTALL_INC_DIR)
diff -ur mercury-srcdist-13.05.2.orig/browser/Mmakefile mercury-srcdist-13.05.2/browser/Mmakefile
--- mercury-srcdist-13.05.2.orig/browser/Mmakefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/browser/Mmakefile	2014-02-18 10:11:16.000000000 +0100
@@ -250,17 +250,29 @@
 
 # there is no browser in the .NET or Erlang backends
 
+.PHONY: libgrade
+libgrade:
+
 .PHONY: install_library
 install_library:
 
 else
 ifeq ($(MMAKE_USE_MMC_MAKE),yes)
 
+.PHONY: libgrade
+libgrade:
+
 .PHONY: install_library
 install_library: lib$(BROWSER_LIB_NAME).install
 
 else 
 
+.PHONY: libgrade
+libgrade: \
+	lib$(BROWSER_LIB_NAME).$A \
+	lib$(BROWSER_LIB_NAME).$(EXT_FOR_SHARED_LIB) \
+	all-ints
+
 .PHONY: install_library
 install_library:				\
 	lib$(BROWSER_LIB_NAME).install_library	\
diff -ur mercury-srcdist-13.05.2.orig/library/Mmakefile mercury-srcdist-13.05.2/library/Mmakefile
--- mercury-srcdist-13.05.2.orig/library/Mmakefile	2013-04-09 11:22:17.000000000 +0200
+++ mercury-srcdist-13.05.2/library/Mmakefile	2014-02-18 10:11:16.000000000 +0100
@@ -514,6 +514,9 @@
 
 # XXX some modules in the browser directory need library.int3
 
+.PHONY: libgrade
+libgrade: mercury.dll
+
 .PHONY: install_library
 install_library: mercury.dll install_grade_dirs install_gac \
 		install_strong_name lib$(STD_LIB_NAME).install_grade_hdrs \
@@ -552,6 +555,9 @@
 
 else
 
+.PHONY: libgrade
+libgrade: lib$(STD_LIB_NAME)
+
 # we depend on lib$(STD_LIB_NAME) because lib$(STD_LIB_NAME).install_library
 # doesn't make library.int3, but some modules in the browser directory need it.
 
@@ -562,6 +568,9 @@
 
 else	#ifneq ($(MMAKE_USE_MMC_MAKE),yes)
 
+.PHONY: libgrade
+libgrade: lib$(STD_LIB_NAME) all-ints
+
 .PHONY: install_mercury
 install_mercury: install_library
 
diff -ur mercury-srcdist-13.05.2.orig/Makefile mercury-srcdist-13.05.2/Makefile
--- mercury-srcdist-13.05.2.orig/Makefile	2013-04-09 11:22:16.000000000 +0200
+++ mercury-srcdist-13.05.2/Makefile	2014-02-18 10:11:16.000000000 +0100
@@ -72,6 +72,6 @@
 	$(MMAKE) uninstall
 
 .DEFAULT:
-	$(MMAKE) $@
+	$(MMAKE) MMAKEFLAGS=$(PARALLEL) $@
 
 #-----------------------------------------------------------------------------#
diff -ur mercury-srcdist-13.05.2.orig/mdbcomp/Mmakefile mercury-srcdist-13.05.2/mdbcomp/Mmakefile
--- mercury-srcdist-13.05.2.orig/mdbcomp/Mmakefile	2013-04-09 11:22:17.000000000 +0200
+++ mercury-srcdist-13.05.2/mdbcomp/Mmakefile	2014-02-18 10:11:16.000000000 +0100
@@ -215,17 +215,29 @@
 
 # there is no debugger in the .NET or Erlang backends
 
+.PHONY: libgrade
+libgrade:
+
 .PHONY: install_library
 install_library:
 
 else
 ifeq ($(MMAKE_USE_MMC_MAKE),yes)
 
+.PHONY: libgrade
+libgrade:
+
 .PHONY: install_library
 install_library: lib$(MDBCOMP_LIB_NAME).install
 
 else
 
+.PHONY: libgrade
+libgrade: \
+	lib$(MDBCOMP_LIB_NAME).$A \
+	lib$(MDBCOMP_LIB_NAME).$(EXT_FOR_SHARED_LIB) \
+	all-ints
+
 .PHONY: install_library
 install_library:				\
 	lib$(MDBCOMP_LIB_NAME).install_library	\
diff -ur mercury-srcdist-13.05.2.orig/Mmakefile mercury-srcdist-13.05.2/Mmakefile
--- mercury-srcdist-13.05.2.orig/Mmakefile	2013-12-02 03:45:42.000000000 +0100
+++ mercury-srcdist-13.05.2/Mmakefile	2014-02-18 10:11:16.000000000 +0100
@@ -321,6 +321,47 @@
 		mdbcomp browser ssdb trace
 	+cd mfilterjavac && $(SUBDIR_MMAKE)
 
+.PHONY: default_grade
+default_grade:
+	+if test "$(DEFAULT_GRADE)" != "$(GRADE)"; then                              \
+	    gc_grade=`scripts/ml --grade $(DEFAULT_GRADE) --print-gc-grade`;         \
+	    IWS=`/bin/pwd`/install_grade_dir.$(DEFAULT_GRADE);                       \
+	    (                                                                        \
+		scripts/prepare_install_dir $${IWS} &&                               \
+		( cd $${IWS}/boehm_gc &&                                             \
+		  $(SUBDIR_MMAKE) MC=mmc GRADE=$(DEFAULT_GRADE) WORKSPACE=$${IWS}    \
+			GC_GRADE=$${gc_grade} libgrade ) &&                          \
+		( cd $${IWS}/runtime &&                                              \
+		  $(SUBDIR_MMAKE) MC=mmc GRADE=$(DEFAULT_GRADE) WORKSPACE=$${IWS}    \
+			libgrade ) &&                                                \
+		( cd $${IWS}/library &&                                              \
+		  $(SUBDIR_MMAKE) MC=mmc GRADE=$(DEFAULT_GRADE) WORKSPACE=$${IWS}    \
+			depend &&                                                    \
+		  $(SUBDIR_MMAKE) MC=mmc GRADE=$(DEFAULT_GRADE) WORKSPACE=$${IWS}    \
+			libgrade ) &&                                                \
+		( cd $${IWS}/mdbcomp &&                                              \
+		  $(SUBDIR_MMAKE) MC=mmc GRADE=$(DEFAULT_GRADE) WORKSPACE=$${IWS}    \
+			depend &&                                                    \
+		  $(SUBDIR_MMAKE) MC=mmc GRADE=$(DEFAULT_GRADE) WORKSPACE=$${IWS}    \
+			libgrade ) &&                                                \
+		( cd $${IWS}/browser &&                                              \
+		  $(SUBDIR_MMAKE) MC=mmc GRADE=$(DEFAULT_GRADE) WORKSPACE=$${IWS}    \
+			depend &&                                                    \
+		  $(SUBDIR_MMAKE) MC=mmc GRADE=$(DEFAULT_GRADE) WORKSPACE=$${IWS}    \
+			libgrade ) &&                                                \
+		( cd $${IWS}/ssdb &&                                                 \
+		  $(SUBDIR_MMAKE) MC=mmc GRADE=$(DEFAULT_GRADE) WORKSPACE=$${IWS}    \
+			depend &&                                                    \
+		  $(SUBDIR_MMAKE) MC=mmc GRADE=$(DEFAULT_GRADE) WORKSPACE=$${IWS}    \
+			libgrade ) &&                                                \
+		( cd $${IWS}/trace &&                                                \
+		  $(SUBDIR_MMAKE) MC=mmc GRADE=$(DEFAULT_GRADE) WORKSPACE=$${IWS}    \
+			libgrade ) &&                                                \
+		true                                                                 \
+	    ) ||                                                                     \
+	    { exit 1; };                                                             \
+	fi;
+
 #-----------------------------------------------------------------------------#
 
 .PHONY: tags
diff -ur mercury-srcdist-13.05.2.orig/runtime/Mmakefile mercury-srcdist-13.05.2/runtime/Mmakefile
--- mercury-srcdist-13.05.2.orig/runtime/Mmakefile	2013-06-28 04:14:42.000000000 +0200
+++ mercury-srcdist-13.05.2/runtime/Mmakefile	2014-02-18 10:11:16.000000000 +0100
@@ -471,6 +471,9 @@
 # mercury/library/Mmakefile.)
 # (Runtime support for the Erlang backend is in the library directory.)
 
+.PHONY: libgrade
+libgrade:
+
 .PHONY: install_headers
 install_headers:
 
@@ -483,6 +486,9 @@
 else
 ifeq ($(findstring il,$(GRADE)),il)
 
+.PHONY: libgrade
+libgrade: $(DOTNET_DLLS)
+
 .PHONY: install_headers
 install_headers:
 
@@ -497,6 +503,9 @@
 
 else
 
+.PHONY: libgrade
+libgrade: lib$(RT_LIB_NAME).$A lib$(RT_LIB_NAME).$(EXT_FOR_SHARED_LIB)
+
 # mercury_conf.h needs to be in a separate directory so it can be
 # overridden by a different configuration.
 .PHONY: install_headers
diff -ur mercury-srcdist-13.05.2.orig/ssdb/Mmakefile mercury-srcdist-13.05.2/ssdb/Mmakefile
--- mercury-srcdist-13.05.2.orig/ssdb/Mmakefile	2013-04-09 11:22:17.000000000 +0200
+++ mercury-srcdist-13.05.2/ssdb/Mmakefile	2014-02-18 10:11:16.000000000 +0100
@@ -248,17 +248,29 @@
 
 # There is no ssdb in the .NET or Erlang backends.
 
+.PHONY: libgrade
+libgrade:
+
 .PHONY: install_library
 install_library:
 
 else
 ifeq ($(MMAKE_USE_MMC_MAKE),yes)
 
+.PHONY: libgrade
+libgrade:
+
 .PHONY: install_library
 install_library: lib$(SSDB_LIB_NAME).install
 
 else 
 
+.PHONY: libgrade
+libgrade: \
+	lib$(SSDB_LIB_NAME).$A \
+	lib$(SSDB_LIB_NAME).$(EXT_FOR_SHARED_LIB) \
+	all-ints
+
 .PHONY: install_library
 install_library:				\
 	lib$(SSDB_LIB_NAME).install_library	\
diff -ur mercury-srcdist-13.05.2.orig/trace/Mmakefile mercury-srcdist-13.05.2/trace/Mmakefile
--- mercury-srcdist-13.05.2.orig/trace/Mmakefile	2013-06-28 04:14:42.000000000 +0200
+++ mercury-srcdist-13.05.2/trace/Mmakefile	2014-02-18 10:11:16.000000000 +0100
@@ -328,6 +328,9 @@
 
 # there is no tracing in the .NET, Java, Erlang backends
 
+.PHONY: libgrade
+libgrade:
+
 .PHONY: install_headers
 install_headers:
 
@@ -336,6 +339,13 @@
 
 else
 
+.PHONY: libgrade
+libgrade: \
+		lib$(TRACE_LIB_NAME).$A \
+		lib$(TRACE_LIB_NAME).$(EXT_FOR_SHARED_LIB) \
+		lib$(EVENTSPEC_LIB_NAME).$A \
+		lib$(EVENTSPEC_LIB_NAME).$(EXT_FOR_SHARED_LIB)
+
 .PHONY: install_headers
 install_headers: $(HDRS) $(LIB_GLOBALS_H)
 	cp `vpath_find $(HDRS) $(LIB_GLOBALS_H)` $(INSTALL_INC_DIR)
