--- lockdev-1.0.3/src/ttylock.h.ORIG	2012-08-20 13:30:11.313446308 +0000
+++ lockdev-1.0.3/src/ttylock.h	2012-08-20 13:31:26.482015968 +0000
@@ -36,10 +36,13 @@
 
 /* API of the library */
 
+/* Commented out because these apparently are upstream breakage.
+   They clash with the "static inline" declarations below.
 int ttylock (char * devname);
 int ttywait (char * devname);
 int ttyunlock (char * devname);
 int ttylocked (char * devname);
+*/
 
 
 static inline int
