--- gprbuild-2016_p20170427191900.ORIG/./Makefile	2017-05-05 00:19:55.501138010 +0000
+++ gprbuild-2016_p20170427191900/./Makefile	2017-05-05 00:26:25.043893063 +0000
@@ -24,9 +24,13 @@
 #   $ make -f <GPRBUILD_TREE>/Makefile setup
 #   $ make -f <GPRBUILD_TREE>/Makefile
 
+.NOTPARALLEL:
+
 HOST    = $(shell gcc -dumpmachine)
 TARGET := $(shell gcc -dumpmachine)
 
+LIBDIR = lib
+
 prefix	      := $(dir $(shell which gnatls))..
 BUILD         = production
 PROCESSORS    = 0
@@ -68,8 +72,8 @@
 # Used to pass extra options to GPRBUILD, like -d for instance
 GPRBUILD_OPTIONS=
 
-BUILDER=gprbuild -p -m $(GTARGET) $(RBD) -j${PROCESSORS} -XBUILD=${BUILD} ${GPRBUILD_OPTIONS}
-LIB_INSTALLER=gprinstall -p -f --target=$(TARGET) $(RBD) --prefix=${prefix}
+BUILDER=gprbuild -p -m -v -R $(GTARGET) $(RBD) -j${PROCESSORS} -XBUILD=${BUILD} ${GPRBUILD_OPTIONS}
+LIB_INSTALLER=gprinstall -p -f -v --target=$(TARGET) $(RBD) --prefix=${prefix}
 INSTALLER=exe/$(LIB_INSTALLER)
 CLEANER=gprclean -q $(RBD)
 
@@ -151,21 +155,22 @@
 	$(LIBGPR_INSTALLER) \
 	   -XLIBRARY_TYPE=static \
 	   -XXMLADA_BUILD=static \
-	   --lib-subdir=lib/gpr/static \
+	   --lib-subdir=${LIBDIR}/gpr/static \
 	   --build-name=static
 
 libgpr.install.static-pic:
 	$(LIBGPR_INSTALLER) \
 	   -XLIBRARY_TYPE=static-pic \
 	   -XXMLADA_BUILD=static-pic \
-	   --lib-subdir=lib/gpr/static-pic \
+	   --lib-subdir=${LIBDIR}/gpr/static-pic \
 	   --build-name=static-pic
 
 libgpr.install.shared:
 	$(LIBGPR_INSTALLER) \
 	   -XLIBRARY_TYPE=relocatable \
 	   -XXMLADA_BUILD=relocatable \
-	   --lib-subdir=lib/gpr/relocatable \
+	   --lib-subdir=${LIBDIR}/gpr/relocatable \
+	   --link-lib-subdir=${LIBDIR} \
 	   --build-name=relocatable
 
 libgpr.uninstall:
@@ -207,16 +212,16 @@
 	-$(CLEANER) $(GPR_GPR) -XBUILD=$(BUILD) \
 		-XLIBRARY_TYPE=relocatable
 endif
-	make -C $(MAKEPREFIX)doc clean
-	make -C $(MAKEPREFIX)examples clean
+	$(MAKE) -C $(MAKEPREFIX)doc clean
+	$(MAKE) -C $(MAKEPREFIX)examples clean
 	rm -fr obj exe makefile.setup
 
 .PHONY: doc examples
 
 doc:
-	make -C $(MAKEPREFIX)doc
+	$(MAKE) -C $(MAKEPREFIX)doc
 
 examples: force
-	make -C $(MAKEPREFIX)examples
+	$(MAKE) -C $(MAKEPREFIX)examples
 
 force:
--- gprbuild-2016_p20170427191900.ORIG/./gpr/gpr.gpr	2017-05-05 00:19:55.529136986 +0000
+++ gprbuild-2016_p20170427191900/./gpr/gpr.gpr	2017-05-05 00:23:52.835459079 +0000
@@ -43,29 +43,30 @@
 
    package Compiler is
       Common_Switches := ("-gnat12", "-gnaty", "-gnatQ", "-gnata");
+      Ada_Flags := Split (external ("ADAFLAGS", ""), " ");
 
       case Bld is
          when "debug" =>
             for Default_Switches ("Ada") use Common_Switches &
-              ("-g", "-gnata", "-gnatVa", "-gnatwaCJI", "-gnatwe", "-gnatyg");
+              ("-g", "-gnata", "-gnatVa", "-gnatwaCJI", "-gnatwe", "-gnatyg") & Ada_Flags;
 
             for Local_Configuration_Pragmas use "debug.adc";
 
          when "coverage" =>
             for Default_Switches ("Ada") use Common_Switches &
-              ("-ftest-coverage", "-fprofile-arcs");
+              ("-ftest-coverage", "-fprofile-arcs") & Ada_Flags;
 
          when "profiling" =>
-            for Default_Switches ("Ada") use Common_Switches & ("-pg", "-g");
+            for Default_Switches ("Ada") use Common_Switches & ("-pg", "-g") & Ada_Flags;
 
          when "production" =>
             for Default_Switches ("Ada") use Common_Switches &
-              ("-O2", "-gnatn", "-gnatws");
+              ("-O2", "-gnatn", "-gnatws") & Ada_Flags;
 
             --  Compile all Ada sources to support symbolic-traceback
 
-            for Switches ("gpr*.ad?") use
-              Compiler'Default_Switches ("Ada") & ("-g1");
+            --for Switches ("gpr*.ad?") use
+            --  Compiler'Default_Switches ("Ada") & ("-g1");
       end case;
    end Compiler;
 
