--- flibs.ORIG/src/datastructures/qsortarray_template.f90	2012-07-04 20:52:22.431822077 +0000
+++ flibs/src/datastructures/qsortarray_template.f90	2012-07-04 21:07:37.400995866 +0000
@@ -33,7 +33,7 @@
   interface
      integer function compare(f,g)
        use _QSORTARRAY_MODULE , only : _QSORTARRAY_TYPE
-       type( _QSORTARRAY_TYPE ) :: f, g
+       type( _QSORTARRAY_TYPE ), intent(in) :: f, g
      end function compare
   end interface
   type ( _QSORTARRAY_TYPE ) , dimension(:), allocatable :: backup
@@ -69,7 +69,7 @@
   interface
      integer function compare ( f , g )
        use _QSORTARRAY_MODULE, only : _QSORTARRAY_TYPE
-       type(_QSORTARRAY_TYPE) :: f, g
+       type(_QSORTARRAY_TYPE), intent(in) :: f, g
      end function compare
   end interface
   integer                                    :: i
