--- flibs.ORIG/src/strings/m_vstring.f90	2012-07-04 20:52:22.447821118 +0000
+++ flibs/src/strings/m_vstring.f90	2012-07-04 21:56:37.643426143 +0000
@@ -754,6 +754,7 @@
     integer :: last
     character ( len = 300 ) :: message
     integer :: status
+    equals = .false.
     call vstring_check_string ( this , "vstring_equals" , status )
     if ( status /= VSTRING_ERROR_OK ) then
        return
@@ -1084,6 +1085,7 @@
     else
        nocase_real = .false.
     endif
+    compare = 0
     call vstring_check_string ( this , "vstring_compare" , status )
     if ( status /= VSTRING_ERROR_OK ) then
        return
@@ -1402,6 +1404,7 @@
     type(t_vstring)   :: substring
     logical :: equals
     integer :: status
+    foundIndex = VSTRING_INDEX_UNKNOWN
     call vstring_check_string ( this , "vstring_first_vstring" , status )
     if ( status /= VSTRING_ERROR_OK ) then
        return
@@ -1475,6 +1478,7 @@
     type(t_vstring)   :: substring
     logical :: equals
     integer :: status
+    foundIndex = VSTRING_INDEX_UNKNOWN
     call vstring_check_string ( this , "vstring_last_vstring" , status )
     if ( status /= VSTRING_ERROR_OK ) then
        return
@@ -1926,6 +1930,7 @@
     !
     ! Initialize
     !
+    match = .false.
     call vstring_check_string ( this , "Check string in vstring_match_vstring." , status )
     if ( status /= VSTRING_ERROR_OK ) then
        return
